app/models/pacbio/pool.rb

```diff
@@ -1,26 +1,18 @@
 # frozen_string_literal: true
 
-# ALIQUOT-CLEANUP
-# - Remove the presence validation on libraries
-# - Remove the library_attributes= method and the library/used_aliquot sync behaviour
-# - Remove the used_aliquots_attributes= method
-# - Remove private methods referencing libraries
-
 module Pacbio
   # Pool
   class Pool < ApplicationRecord
     include Aliquotable
     belongs_to :tube, default: -> { Tube.new }
-    has_many :libraries, class_name: 'Pacbio::Library', foreign_key: :pacbio_pool_id,
-                         dependent: :destroy, inverse_of: :pool
-    has_many :well_pools, class_name: 'Pacbio::WellPool', foreign_key: :pacbio_pool_id,
-                          dependent: :nullify, inverse_of: :pool
-    has_many :wells, class_name: 'Pacbio::Well', through: :well_pools
-    has_many :requests, through: :libraries
 
-    # This is dependent on the requests association, so needs to be included
+    has_many :wells, through: :derived_aliquots, source: :used_by, source_type: 'Pacbio::Well'
+    has_many :requests, through: :used_aliquots, source: :source, source_type: 'Pacbio::Request'
+    has_many :libraries, through: :used_aliquots, source: :source, source_type: 'Pacbio::Library'
+
+    # This is dependent on the requests and libraries associations, so needs to be included
     # after that is defined
-    include DualSourcedPool
+    include MultiSourcedPool
```

Comment:
I have some questions about MultiSourcePool,  mostly stemming from my ignorance about Rail concepts. I will ask you offline. :)